pkg/kubelet/dockertools/kube_docker_client.go

```diff
@@ -177,9 +198,13 @@ func (d *kubeDockerClient) PullImage(image string, auth dockertypes.AuthConfig,
 	if err != nil {
 		return err
 	}
-	opts.ImageID = image
+	ctx, cancel := getDefaultContext()
+	defer cancel()
 	opts.RegistryAuth = base64Auth
-	resp, err := d.client.ImagePull(getDefaultContext(), opts, nil)
+	resp, err := d.client.ImagePull(ctx, image, opts)
```

Comment:
I'd expect pulling to frequently take longer than 2 minutes.
Maybe a longer timeout?